### PR TITLE
Bluetooth: Controller: Use sys_get_le24 to fetch 24-bit data 

### LINKED ISSUE #27971: Bluetooth: controller: use sys_get_le24 to fetch 24-bit data
**Is your enhancement proposal related to a problem? Please describe.**
Refer to https://github.com/zephyrproject-rtos/zephyr/pull/27357#discussion_r477324409
and
https://github.com/zephyrproject-rtos/zephyr/pull/27357#discussion_r477319846

**Describe the solution you'd like**
- Use sys_get_le24() to pass crc_init value
- Define CRC polynomial and use the define instead of magic numbers.

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by adpu88**
Use sys_get_le24 to pass CRC init value.
Define RADIO_CRC_POLYNOMIAL to replace the magic number.

Fixes #27971.

Signed-off-by: Adhipta Putra <adhipta.putra@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.h [hunk 1]
```diff
@@ -5,6 +5,9 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define RADIO_CRC_POLYNOMIAL ((0x5bUL) | ((0x06UL) << 8) | ((0x00UL) << 16))
+
+
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.h:10]: Define this in the `subsys/bluetooth/controller/ll_sw/pdu.h` as `PDU_CRC_POLYNOMIAL`

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c [hunk 2]
```diff
@@ -367,7 +367,7 @@ static int common_prepare_cb(struct lll_prepare_param *p, bool is_resume)
 
 	aa = sys_cpu_to_le32(PDU_AC_ACCESS_ADDR);
 	radio_aa_set((uint8_t *)&aa);
-	radio_crc_configure(((0x5bUL) | ((0x06UL) << 8) | ((0x00UL) << 16)),
+	radio_crc_configure(RADIO_CRC_POLYNOMIAL,
 			    0x555555);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c:371]: `0x555555` could also be defined in `pdu.h`

**Review by cvinayak — CHANGES_REQUESTED** [review]
Not verified if all instances of 24-bit use are covered in this PR, otherwise minor changes request.

Do rebase to resolve filename (change) conflicts.

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.h (hunk 1 above)
Comment (adpu88) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.h:10]: Will do

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c (hunk 2 above)
Comment (adpu88) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c:371]: I will define it as PDU_AC_CRC_IV in pdu.h as well

**Review by cvinayak — APPROVED** [review]

**Review by thoh-ot — APPROVED** [review]

**Comment by carlescufi** [comment]
@adpu88 could you please fix the whitelist errors?

**Review by carlescufi — APPROVED** [review]

**Comment by cvinayak** [comment]
Commit 2bad17bbe8:
1: UC5 Title exceeds max length (76>75): "Bluetooth: Controller: Replaced RADIO_CRC_POLYNOMIAL with PDU_CRC_POLYNOMIAL"
2: B4 Second line is not empty: "Replaced 0x555555 with PDU_AC_CRC_IV"

**Pull request merged**